ams/server/src/main/java/com/netease/arctic/server/optimizing/plan/CommonPartitionEvaluator.java

```diff
@@ -159,7 +159,7 @@ public boolean fileShouldRewrite(DataFile dataFile, List<ContentFile<?>> deletes
     if (isFragmentFile(dataFile)) {
       return true;
     }
-    return getRecordCount(deletes) > dataFile.recordCount() * config.getMajorDuplicateRatio();
+    return getPosDeletesRecordCount(deletes) > dataFile.recordCount() * config.getMajorDuplicateRatio();
```

Comment:
Can you add some comments here to explain why only the count of positional deletes is calculated?

Reply:
Done.